demos/python_demos/object_detection_demo_faceboxes/detector.py

```diff
@@ -0,0 +1,200 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import itertools
+import math
+import numpy as np
+import os
+import cv2
+
+from openvino.inference_engine import IENetwork
+
+class Detector(object):
+    def __init__(self, ie, model_path, device='CPU'):
+        model = IENetwork(model=model_path, weights=os.path.splitext(model_path)[0] + '.bin')
+
+        assert len(model.inputs) == 1, "Expected 1 input blob"
+        assert len(model.outputs) == 2, "Expected 2 output blobs"
+
+        self._input_layer_name = next(iter(model.inputs))
+        self._output_layer_names = sorted(model.outputs)
+
+        self._ie = ie
+        self._exec_model = self._ie.load_network(model, device)
+        self.infer_time = -1
+        _, channels, self.input_height, self.input_width = model.inputs[self._input_layer_name].shape
+        assert channels == 3, "Expected 3-channel input"
+
+        self.min_sizes = [[32, 64, 128], [256], [512]]
+        self.steps = [32, 64, 128]
+        self.variance = [0.1, 0.2]
+        self.confidence_threshold = 0.05
```

Comment:
do you really need so small threshold and so large topk for demo purposes?

Reply:
How should I choose these parameters for demo? Are there any default values?

Reply:
These values ​​were taken from the repository with the original model PyTorch as default values(https://github.com/zisianw/FaceBoxes.PyTorch/blob/master/test.py#L23). I also found the value for confidence_threshold for demo in the repository with Caffe model(https://github.com/sfzhang15/FaceBoxes/blob/master/test/demo.py#L37).

Reply:
For PyTorch there are two thresholds: `confidence_threshold` and `vis_thres`. In your case these thresholds seems to be merged (which I think is fine). But the resulting threshold for PyTorch is `vis_thres` which is 0.5.

You should set the value to 0.5. Sores for labels are interpreted as probabilities usually, so 0.5 is an edge when it becomes more probable that the box is a suitable detection rather than background. If you want, you can also add an argument for the demo to allow changing that value.

Reply:
Set default value for confidence_threshold to 0.5. added argument to change it